### PR TITLE
BOAC-4363: Added methods to fetch and transform YCBM data to json

### PR DESCRIPTION
**Pull request by cesarvh**
https://jira.ets.berkeley.edu/jira/browse/BOAC-4363

### CODE REVIEW

File: boac/externals/data_loch.py [hunk 1]
```diff
@@ -657,6 +658,14 @@ def get_sis_advising_appointments(sid):
         ORDER BY created_at, updated_at, id"""
     return safe_execute_rds(sql, sid=sid)
 
+def get_ycbm_advising_appointments(sid):
+    sql = f"""
+        SELECT
+            id, student_sid, title, starts_at, ends_at, cancelled, cancellation_reason, advisor_name, appointment_type, details
+        FROM {advising_appointments_schema()}.ycbm_advising_appointments
+        WHERE student_sid=:sid
+        ORDER BY id"""
```
Comment (johncrossman) [inline boac/externals/data_loch.py:667]: @cesarvh - does the order by matter? If yes, then I'd expect `ORDER BY created_at, updated_at`. Thoughts?

File: boac/merged/advising_appointment.py [hunk 2]
```diff
@@ -159,8 +171,11 @@ def appointment_to_compatible_json(appointment, topics=(), attachments=None, eve
     advisor_uid = appointment.get('advisor_uid')
     appointment_id = appointment.get('id')
     appointment_type = appointment.get('appointment_type')
+    cancelled = appointment.get('cancelled')
     departments = []
     dept_codes = appointment.get('advisor_dept_codes') or []
+    created_by = appointment.get('created_by') if appointment.get('created_by') else 'YCBM'
```
Comment (johncrossman) [inline boac/merged/advising_appointment.py:177]: @cesarvh 
* Will front-end know what to do with created_by=YCBM? (created_by is typically a UID)
* Is `appointment.get('created_by') or 'YCBM'` a more concise equivalent?

File: boac/merged/advising_appointment.py [hunk 3]
```diff
@@ -198,6 +213,11 @@ def appointment_to_compatible_json(appointment, topics=(), attachments=None, eve
             'studentContactInfo': appointment.get('student_contact_info'),
             'studentContactType': appointment.get('student_contact_type'),
         })
+    if cancelled: 
+        api_json.update({
+            'cancelled': True,
+            'cancelReason': appointment.get('cancellation_reason')
+        })
```
Comment (johncrossman) [inline boac/merged/advising_appointment.py:220]: @cesarvh - I suggest the api JSON have consistent structure. In this case, it means the two props above should always be in the payload. We'd change it to:

```
'cancelled':  cancelled if created_by == 'YCBM' else None,
'cancelReason': appointment.get('cancellation_reason') if created_by == 'YCBM' else None,
```

**Comment by cesarvh** [comment]
@johncrossman  Thanks for the feedback! For the query, order shouldn't matter. We don't have created_at or updated_at info for YCBM appointments, just the start time of the appointment

**Comment by cesarvh** [comment]
Hmm... trying to figure out the travis failure here... ignore for the time being

**Comment by johncrossman** [comment]
Thanks @cesarvh   The front-end code (eg, `AcademicTimelineTable.vue`) assumes that notes and appointments have 'createdAt' and 'updatedAt'.  Things may (or may not?) break when those values are absent.

**Comment by johncrossman** [comment]
@cesarvh - I think BOA's `loch.sql` (used during tests to mock the data lake) needs `ycbm_advising_appointments`

**Comment by cesarvh** [comment]
@johncrossman Yep! Forgot to create the schema on that last push. Hopefully this one does it 🙃🙃🙃

**Comment by cesarvh** [comment]
@johncrossman should be all good now :)

File: tests/test_merged/test_advising_appointment.py [hunk 4]
```diff
@@ -54,7 +54,7 @@ def test_get_advising_appointments(self, app, fake_auth):
         assert appointments[0]['appointmentType'] is None
         assert len(appointments[0]['attachments']) == 1
         assert appointments[0]['createdAt'] == '2017-10-31T12:00:00+00:00'
-        assert appointments[0]['createdBy'] is None
+        assert appointments[0]['createdBy'] is None or 'YCBM'
```
Comment (johncrossman) [inline tests/test_merged/test_advising_appointment.py:57]: @cesarvh - I think the assertion above will always be true. Do you want the following instead?
```
assert appointments[0]['createdBy'] in [None, 'YCBM']
```

**Comment by johncrossman** [comment]
Thanks @cesarvh  I'm waiting on Travis.

**Pull request merged**